launcher/ui/MainWindow.cpp

```diff
@@ -1618,20 +1615,61 @@ void MainWindow::on_actionCreateInstanceShortcut_triggered()
     return;
 #endif
     args.append({ "--launch", m_selectedInstance->id() });
-    if (FS::createShortcut(desktopFilePath, appPath, args, m_selectedInstance->name(), iconPath)) {
-#if not defined(Q_OS_MACOS)
-        QMessageBox::information(this, tr("Create instance shortcut"), tr("Created a shortcut to this instance on your desktop!"));
-#else
-        QMessageBox::information(this, tr("Create instance shortcut"), tr("Created a shortcut to this instance!"));
-#endif
-    } else {
+
+    if (!FS::createShortcut(std::move(shortcutFilePath), appPath, args, m_selectedInstance->name(), iconPath)) {
 #if not defined(Q_OS_MACOS)
         iconFile.remove();
 #endif
         QMessageBox::critical(this, tr("Create instance shortcut"), tr("Failed to create instance shortcut!"));
+        return;
     }
 }
 
+void MainWindow::on_actionCreateInstanceShortcutOther_triggered() {
+        if (!m_selectedInstance)
+                return;
+
+        QString defaultedDir = FS::getDesktopDir();
+#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
+        QString extension = ".desktop";
+#elif defined(Q_OS_WINDOWS)
+        QString extension = ".lnk";
+#else
+        QString extension = "";
+#endif
+
+        QString shortcutFilePath = FS::PathCombine(defaultedDir, FS::RemoveInvalidFilenameChars(m_selectedInstance->name()) + extension);
+        QFileDialog fileDialog;
+        // workaround to make sure the portal file dialog opens in the desktop directory
+        fileDialog.setDirectoryUrl(defaultedDir);
+
+        shortcutFilePath = fileDialog.getSaveFileName(this, tr("Create Shortcut"), shortcutFilePath, tr("Desktop Entries") + " (*.desktop)");
```

Comment:
Here the .desktop should be replaced with the specific extension for that operating system